source/common/http/conn_pool_grid.cc

```diff
@@ -260,6 +296,10 @@ absl::optional<ConnectivityGrid::PoolIterator> ConnectivityGrid::nextPool(PoolIt
   return createNextPool();
 }
 
+bool ConnectivityGrid::isPoolHttp3(const ConnectionPool::Instance& pool) {
+  return &pool == pools_.begin()->get();
```

Comment:
Comparing pointer values can be brittle, any way we could instead add a bool on the pool or something?

Reply:
Discussed offline that given the pools are handed out to the connection attempts via pointer it's reasonably safe to use them here. So keeping code as is.